### ISSUE TITLE
Server/ Video Chat connectivity issues

### ISSUE THREAD

**Issue by shalem93** [QuickBlox/quickblox-ios-sdk#505]
_Help avoid duplicate issue reports, check [existing issues](https://github.com/QuickBlox/quickblox-ios-sdk/issues)_

**Environment details**
_iOS version, Quickblox iOS SDK version, QuickbloxWebRTC SDK version (optional)_
iOS 9.3, QB RTC 2.2

**Did this work before?**
No

**Expected behavior**
Users with LTE and Wi-Fi are able to access our servers and video chat with each other

**Actual behavior**
LTE users can barely use our app, they can't vid chat with other users most times. Wi-fi works maybe 50% of the time. We are based in the US and same with all our users so that may be a big issue. Is there a way we can make the connection for our users better???

**Logs**
_Attach full logs that will describe your problem (please, use github [gist](https://gist.github.com/))_

**Steps to reproduce the behavior**
Attempt to use our app with LTE instead of WiFi

**Any others comments?**

 Just want some helpful responses so we can get our users connected much better!

**Comment by Raikerian** [comment]
Hi,

Do you have problem with audio/video calls or chat service in general?

**Comment by shalem93** [comment]
Yes we are having problems with audio and video calls. Depending on a persons strength of connection, some aren't receiving video feeds, yet the person they are talking to can see them. That is a clear problem. Some people just can't connect to other people. And everyone had good signal, but it seems you need to have near perfect to use Quickblox which shouldn't be the case! It makes our app very difficult for the common man to use, and we need to fix this so everyone can use it :)

**Comment by shalem93** [comment]
Any updates??

**Comment by Raikerian** [comment]
Please enable stat reports and send us logs if possible, use this delegate implementation as an example:
https://github.com/QuickBlox/q-municate-ios/blob/master/Q-municate/Classes/Core/QMCallManager/QMCallManager.m#L184

And add this to appdelegate as quickblox settings:
```objc
[QBRTCConfig setStatsReportTimeInterval:1.0f]; // set to 1.0f to enable stats report
```

**Comment by shalem93** [comment]
Ok ill get a good example of stats reporting of LTE that doesn't work very well in approx 12 hours. Its late here and it will take time to reenact this problem. Thx for sharing, this should give us more detail about what's occurring, not necessarily how to fix the issue though...

**Comment by shalem93** [comment]
Its hard getting stats because you need really good internet signal to access the app and you can't get stats reported if the app isn't accessible. Have you heard of any users from the US having any connectivity issues?

**Comment by amitk-virtueinfo** [comment]
Hi I'm doing video call by web to iPhone 6 
I'm not able to see video from iphone to web but it's working fine from web to iOS on my iphone both video render perfectly but not on web, if i call web to web it's working fine.

Same issues on chat message, messages recived from web but I can't send message from ios app using dialog 

QBRTCVideoFormat *videoFormat = [[QBRTCVideoFormat alloc] init];
    videoFormat.frameRate = 10;
    videoFormat.pixelFormat = QBRTCPixelFormat420f;
    videoFormat.width = 320;
    videoFormat.height = 320;
    
    self.videoCapture = [[QBRTCCameraCapture alloc] initWithVideoFormat:videoFormat position:AVCaptureDevicePositionFront]; // or AVCaptureDevicePositionBack

    self.videoCapture.previewLayer.frame = self.localVideoFrame.bounds;
    [self.videoCapture startSession];

...
...

    self.chatDialog = [[QBChatDialog alloc] initWithDialogID:nil type:QBChatDialogTypePrivate];
    self.chatDialog.name = @"";
    self.chatDialog.occupantIDs = @[userID];
    // Handling 'typing' status.
    [self.chatDialog setOnUserIsTyping:^(NSUInteger userID) {
        if ([QBSession currentSession].currentUser.ID == userID) {
            return;
        }
        NSLog(@"%s ------------------ SA_START_TYPING",__FUNCTION__);
    }];
    
    // Handling user stopped typing.
    [self.chatDialog setOnUserStoppedTyping:^(NSUInteger userID) {
        if ([QBSession currentSession].currentUser.ID == userID) {
            return;
        }
        NSLog(@"%s ------------------------------------ SA_STOP_TYPING",__FUNCTION__);
    }];

    [QBRequest createDialog:self.chatDialog successBlock:^(QBResponse *response, QBChatDialog *createdDialog) {
        NSLog(@"%s Created dialog: %@",__FUNCTION__, createdDialog);
    } errorBlock:^(QBResponse *response) {
        NSLog(@"%s Failed to create dialog with error: %@",__FUNCTION__, response.error);
    }];

any solution how to solve it ?

Here is Log : 
CN 7ms | local->local/udp | (s)38Kbps | (r)889Kbps
VS (input) 0x0@0fps | (sent) 0x0@0fps
VS (enc) 0bps/0bps | (sent) 0bps/300Kbps | 0ms | VP8
VR (recv) 640x480@16fps | (decoded)17 | (output)17fps | 849Kbps/1.03Mbps | 18ms
AS 36Kbps | opus
AR 39Kbps | opus | 95ms | (expandrate)0
2016-07-08 03:56:32.250 Olc [29440:2228382] (cpu)69%
CN 5ms | local->local/udp | (s)38Kbps | (r)871Kbps
VS (input) 0x0@0fps | (sent) 0x0@0fps
VS (enc) 0bps/0bps | (sent) 0bps/300Kbps | 0ms | VP8
VR (recv) 640x480@16fps | (decoded)17 | (output)17fps | 839Kbps/1.11Mbps | 16ms
AS 36Kbps | opus
AR 38Kbps | opus | 90ms | (expandrate)0
2016-07-08 03:56:33.237 Olc [29440:2228382] (cpu)68%
CN 4ms | local->local/udp | (s)38Kbps | (r)951Kbps
VS (input) 0x0@0fps | (sent) 0x0@0fps
VS (enc) 0bps/0bps | (sent) 0bps/300Kbps | 0ms | VP8
VR (recv) 640x480@15fps | (decoded)15 | (output)15fps | 872Kbps/1.19Mbps | 13ms
AS 36Kbps | opus
AR 40Kbps | opus | 89ms | (expandrate)0
2016-07-08 03:56:34.252 Olc [29440:2228382] (cpu)65%
CN 3ms | local->local/udp | (s)37Kbps | (r)865Kbps
VS (input) 0x0@0fps | (sent) 0x0@0fps
VS (enc) 0bps/0bps | (sent) 0bps/300Kbps | 0ms | VP8
VR (recv) 640x480@14fps | (decoded)14 | (output)14fps | 820Kbps/1.24Mbps | 13ms
AS 35Kbps | opus
AR 38Kbps | opus | 93ms | (expandrate)0
2016-07-08 03:56:35.248 Olc [29440:2228382] (cpu)76%
CN 3ms | local->local/udp | (s)38Kbps | (r)1.28Mbps
VS (input) 0x0@0fps | (sent) 0x0@0fps
VS (enc) 0bps/0bps | (sent) 0bps/300Kbps | 0ms | VP8
VR (recv) 640x480@13fps | (decoded)16 | (output)16fps | 1.24Mbps/1.33Mbps | 6ms
AS 36Kbps | opus
AR 39Kbps | opus | 96ms | (expandrate)0

**Comment by shalem93** [comment]
My binary just got rejected from Apple because Apple guidelines state the app must work even with poor connection, but I feel our app can't communicate with Quickblox in these circumstances. Any suggestions on how to communicate better with quickblox? Would appreciate speedy help! Below is our crash log Apple provided...

"Performance - 2.1


Your app crashes on iPad and iPhone running iOS 9.3.2 when we sign in.

This occurred when your app was used: 

- Offline
- On Wi-Fi

We have attached detailed crash logs to help troubleshoot this issue."

Incident Identifier: B5A9B67B-2A2F-4907-B51A-CA6EE74B0736
CrashReporter Key:   1b24324ed9cbf80b24b3cdc1f67f3fac21f8e4f2
Hardware Model:      xxx
Process:             XpandU [1817]
Path:                /private/var/containers/Bundle/Application/2C189935-3E21-41A9-8B09-E27154D87E5A/XpandU.app/XpandU
Identifier:          com.zhs2.Xpandu
Version:             2.0 (2.7.5)
Code Type:           ARM-64 (Native)
Parent Process:      launchd [1]

Date/Time:           2016-07-11 18:47:59.59 -0700
Launch Time:         2016-07-11 18:47:57.57 -0700
OS Version:          iOS 9.3.2 (13F69)
Report Version:      105

Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Triggered by Thread:  0

Filtered syslog:
None found

Last Exception Backtrace:
(0x180adadb0 0x18013ff80 0x1809bb098 0x1000835dc 0x185c30c40 0x185cf1230 0x185cf1108 0x185cf0350 0x185cefef4 0x185cefb1c 0x185cefa84 0x185c2c1e4 0x1835be994 0x1835b95d0 0x1835b9490 0x1835b8ac0 0x1835b8820 0x1835b1de4 0x180a90728 0x180a8e4cc 0x180a8e8fc 0x1809b8c50 0x1822a0088 0x185c9a088 0x100058790 0x1805568b8)

Global Trace Buffer (reverse chronological seconds):
6148914685.166306 CFNetwork                 	0x000000018116151c TCP Conn 0x127777460 started
6148914685.624789 CFNetwork                 	0x00000001810bda80 TCP Conn 0x1276abbf0 SSL Handshake DONE
6148914685.706717 CFNetwork                 	0x00000001810bd990 TCP Conn 0x1276abbf0 starting SSL negotiation
6148914685.707175 CFNetwork                 	0x000000018115ff18 TCP Conn 0x1276abbf0 complete. fd: 34, err: 0
6148914685.707545 CFNetwork                 	0x0000000181161444 TCP Conn 0x1276abbf0 event 1. err: 0
6148914685.792113 CFNetwork                 	0x000000018116151c TCP Conn 0x1276abbf0 started
6148914685.925794 CFNetwork                 	0x00000001810bda80 TCP Conn 0x1275a54a0 SSL Handshake DONE
6148914685.926626 CFNetwork                 	0x00000001810bda80 TCP Conn 0x1276b19f0 SSL Handshake DONE
6148914686.007664 CFNetwork                 	0x00000001810bd990 TCP Conn 0x1276b19f0 starting SSL negotiation
6148914686.008374 CFNetwork                 	0x00000001810bd990 TCP Conn 0x1275a54a0 starting SSL negotiation
6148914686.008511 CFNetwork                 	0x000000018115ff18 TCP Conn 0x1276b19f0 complete. fd: 31, err: 0
6148914686.009028 CFNetwork                 	0x000000018115ff18 TCP Conn 0x1275a54a0 complete. fd: 30, err: 0
6148914686.010228 CFNetwork                 	0x0000000181161444 TCP Conn 0x1276b19f0 event 1. err: 0
6148914686.010656 CFNetwork                 	0x0000000181161444 TCP Conn 0x1275a54a0 event 1. err: 0
6148914686.101028 CFNetwork                 	0x000000018116151c TCP Conn 0x1275a54a0 started
6148914686.101332 CFNetwork                 	0x000000018116151c TCP Conn 0x1276b19f0 started
6148914687.325383 CFNetwork                 	0x000000018116151c TCP Conn 0x127694730 started
6148914687.517206 CFNetwork                 	0x00000001810bda80 TCP Conn 0x127680aa0 SSL Handshake DONE
6148914687.521943 CFNetwork                 	0x00000001810bda80 TCP Conn 0x1277374f0 SSL Handshake DONE
6148914687.545909 CFNetwork                 	0x00000001810bda80 TCP Conn 0x12773af30 SSL Handshake DONE
6148914687.546135 CFNetwork                 	0x00000001810bda80 TCP Conn 0x127566620 SSL Handshake DONE
6148914687.684897 CFNetwork                 	0x00000001810bd990 TCP Conn 0x127680aa0 starting SSL negotiation
6148914687.685106 CFNetwork                 	0x000000018115ff18 TCP Conn 0x127680aa0 complete. fd: 20, err: 0
6148914687.685307 CFNetwork                 	0x0000000181161444 TCP Conn 0x127680aa0 event 1. err: 0
6148914687.692132 CFNetwork                 	0x00000001810bd990 TCP Conn 0x1277374f0 starting SSL negotiation
6148914687.692287 CFNetwork                 	0x000000018115ff18 TCP Conn 0x1277374f0 complete. fd: 23, err: 0
6148914687.692536 CFNetwork                 	0x0000000181161444 TCP Conn 0x1277374f0 event 1. err: 0
6148914687.726200 CFNetwork                 	0x00000001810bd990 TCP Conn 0x127566620 starting SSL negotiation
6148914687.726359 CFNetwork                 	0x00000001810bd990 TCP Conn 0x12773af30 starting SSL negotiation
6148914687.726434 CFNetwork                 	0x000000018115ff18 TCP Conn 0x127566620 complete. fd: 22, err: 0
6148914687.726749 CFNetwork                 	0x000000018115ff18 TCP Conn 0x12773af30 complete. fd: 12, err: 0
6148914687.726993 CFNetwork                 	0x0000000181161444 TCP Conn 0x127566620 event 1. err: 0
6148914687.727149 CFNetwork                 	0x0000000181161444 TCP Conn 0x12773af30 event 1. err: 0
6148914687.766575 CFNetwork                 	0x000000018116151c TCP Conn 0x127680aa0 started
6148914687.774636 CFNetwork                 	0x000000018116151c TCP Conn 0x1277374f0 started
6148914687.816304 CFNetwork                 	0x000000018116151c TCP Conn 0x12773af30 started
6148914687.819542 CFNetwork                 	0x000000018116151c TCP Conn 0x127566620 started
6148914687.825997 CFNetwork                 	0x00000001811bbe5c Creating default cookie storage with default identifier
6148914687.825997 CFNetwork                 	0x00000001811bbe28 Faulting in CFHTTPCookieStorage singleton
6148914687.825998 CFNetwork                 	0x000000018120de50 Faulting in NSHTTPCookieStorage singleton



Thread 0 name:  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x000000018067411c 0x180658000 + 114972
1   libsystem_pthread.dylib       	0x0000000180740ef8 0x18073c000 + 20216
2   libsystem_c.dylib             	0x00000001805e5dac 0x180584000 + 400812
3   libc++abi.dylib               	0x00000001801193f4 0x180118000 + 5108
4   libc++abi.dylib               	0x0000000180135e98 0x180118000 + 122520
5   libobjc.A.dylib               	0x0000000180140248 0x180138000 + 33352
6   libc++abi.dylib               	0x0000000180132f44 0x180118000 + 110404
7   libc++abi.dylib               	0x0000000180132b10 0x180118000 + 109328
8   libobjc.A.dylib               	0x0000000180140120 0x180138000 + 33056
9   CoreFoundation                	0x00000001809b8cf8 0x1809b0000 + 36088
10  GraphicsServices              	0x00000001822a0088 0x182294000 + 49288
11  UIKit                         	0x0000000185c9a088 0x185c1c000 + 516232
12  XpandU                        	0x0000000100058790 0x100048000 + 67472
13  libdyld.dylib                 	0x00000001805568b8 0x180554000 + 10424

Thread 1:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 2 name:  Dispatch queue: com.apple.libdispatch-manager
Thread 2:
0   libsystem_kernel.dylib        	0x00000001806754d8 0x180658000 + 120024
1   libdispatch.dylib             	0x00000001805387d8 0x180524000 + 83928
2   libdispatch.dylib             	0x0000000180527648 0x180524000 + 13896

Thread 3:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 4:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 5 name:  Thread 0x0x1275390e0
Thread 5:
0   libsystem_kernel.dylib        	0x0000000180674344 0x180658000 + 115524
1   XpandU                        	0x0000000100412088 0x100048000 + 3973256
2   XpandU                        	0x0000000100404488 0x100048000 + 3916936
3   XpandU                        	0x0000000100419aa0 0x100048000 + 4004512
4   XpandU                        	0x00000001004198c4 0x100048000 + 4004036
5   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
6   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
7   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 6 name:  Thread 0x0x127539a40
Thread 6:
0   libsystem_kernel.dylib        	0x0000000180674344 0x180658000 + 115524
1   XpandU                        	0x0000000100412088 0x100048000 + 3973256
2   XpandU                        	0x0000000100404488 0x100048000 + 3916936
3   XpandU                        	0x0000000100419aa0 0x100048000 + 4004512
4   XpandU                        	0x00000001004198c4 0x100048000 + 4004036
5   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
6   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
7   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 7 name:  SCTP iterator
Thread 7:
0   libsystem_kernel.dylib        	0x0000000180673f24 0x180658000 + 114468
1   libsystem_pthread.dylib       	0x000000018073ece8 0x18073c000 + 11496
2   XpandU                        	0x00000001004fb32c 0x100048000 + 4928300
3   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
4   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
5   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 8 name:  SCTP timer
Thread 8:
0   libsystem_kernel.dylib        	0x0000000180674344 0x180658000 + 115524
1   XpandU                        	0x00000001004fb7b0 0x100048000 + 4929456
2   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
3   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
4   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 9 name:  VoiceProcessThread
Thread 9:
0   libsystem_kernel.dylib        	0x0000000180673f24 0x180658000 + 114468
1   libsystem_pthread.dylib       	0x000000018073ece8 0x18073c000 + 11496
2   XpandU                        	0x000000010041eb60 0x100048000 + 4025184
3   XpandU                        	0x00000001004f2ad8 0x100048000 + 4893400
4   XpandU                        	0x000000010063da04 0x100048000 + 6248964
5   XpandU                        	0x000000010042004c 0x100048000 + 4030540
6   XpandU                        	0x000000010041fff4 0x100048000 + 4030452
7   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
8   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
9   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 10 name:  AVAudioSession Notify Thread
Thread 10:
0   libsystem_kernel.dylib        	0x0000000180658fd8 0x180658000 + 4056
1   libsystem_kernel.dylib        	0x0000000180658e54 0x180658000 + 3668
2   CoreFoundation                	0x0000000180a90c60 0x1809b0000 + 920672
3   CoreFoundation                	0x0000000180a8e964 0x1809b0000 + 911716
4   CoreFoundation                	0x00000001809b8c50 0x1809b0000 + 35920
5   libAVFAudio.dylib             	0x00000001871299e0 0x1870e4000 + 285152
6   libAVFAudio.dylib             	0x00000001870fe75c 0x1870e4000 + 108380
7   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
8   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
9   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 11:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 12:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 13:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 14 name:  com.apple.NSURLConnectionLoader
Thread 14:
0   libsystem_kernel.dylib        	0x0000000180658fd8 0x180658000 + 4056
1   libsystem_kernel.dylib        	0x0000000180658e54 0x180658000 + 3668
2   CoreFoundation                	0x0000000180a90c60 0x1809b0000 + 920672
3   CoreFoundation                	0x0000000180a8e964 0x1809b0000 + 911716
4   CoreFoundation                	0x00000001809b8c50 0x1809b0000 + 35920
5   CFNetwork                     	0x0000000181139c68 0x18108c000 + 711784
6   Foundation                    	0x00000001814afe4c 0x1813bc000 + 998988
7   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
8   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
9   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 15:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 16 name:  Dispatch queue: NSOperationQueue 0x1275ec940 :: NSOperation 0x1276e7320 (QOS: LEGACY)
Thread 16:
0   libsystem_kernel.dylib        	0x0000000180659014 0x180658000 + 4116
1   libdispatch.dylib             	0x00000001805363e8 0x180524000 + 74728
2   CoreMotion                    	0x000000018697f73c 0x18692c000 + 341820
3   CoreMotion                    	0x00000001869637e4 0x18692c000 + 227300
4   Foundation                    	0x0000000181494540 0x1813bc000 + 886080
5   Foundation                    	0x00000001813e6870 0x1813bc000 + 174192
6   Foundation                    	0x00000001813d6e48 0x1813bc000 + 110152
7   Foundation                    	0x0000000181496934 0x1813bc000 + 895284
8   libdispatch.dylib             	0x000000018052547c 0x180524000 + 5244
9   libdispatch.dylib             	0x00000001805314c0 0x180524000 + 54464
10  libdispatch.dylib             	0x0000000180528f80 0x180524000 + 20352
11  libdispatch.dylib             	0x0000000180533390 0x180524000 + 62352
12  libdispatch.dylib             	0x00000001805330b0 0x180524000 + 61616
13  libsystem_pthread.dylib       	0x000000018073d470 0x18073c000 + 5232
14  libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 17:
0   libsystem_kernel.dylib        	0x0000000180674b48 0x180658000 + 117576
1   libsystem_pthread.dylib       	0x000000018073d530 0x18073c000 + 5424
2   libsystem_pthread.dylib       	0x000000018073d020 0x18073c000 + 4128

Thread 18 name:  com.apple.CFSocket.private
Thread 18:
0   libsystem_kernel.dylib        	0x0000000180674344 0x180658000 + 115524
1   CoreFoundation                	0x0000000180a971c8 0x1809b0000 + 946632
2   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
3   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
4   libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 19 name:  GCDAsyncSocket-CFStream
Thread 19:
0   libsystem_kernel.dylib        	0x0000000180658fd8 0x180658000 + 4056
1   libsystem_kernel.dylib        	0x0000000180658e54 0x180658000 + 3668
2   CoreFoundation                	0x0000000180a90c60 0x1809b0000 + 920672
3   CoreFoundation                	0x0000000180a8e964 0x1809b0000 + 911716
4   CoreFoundation                	0x00000001809b8c50 0x1809b0000 + 35920
5   Foundation                    	0x00000001813c8cfc 0x1813bc000 + 52476
6   XpandU                        	0x00000001001ba510 0x100048000 + 1516816
7   Foundation                    	0x00000001814afe4c 0x1813bc000 + 998988
8   libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
9   libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
10  libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 20 name:  com.apple.CoreMotion.MotionThread
Thread 20:
0   libsystem_kernel.dylib        	0x0000000180658fd8 0x180658000 + 4056
1   libsystem_kernel.dylib        	0x0000000180658e54 0x180658000 + 3668
2   IOKit                         	0x0000000180d8172c 0x180d4c000 + 218924
3   IOKit                         	0x0000000180d77dbc 0x180d4c000 + 179644
4   IOKit                         	0x0000000180d767f4 0x180d4c000 + 174068
5   CoreMotion                    	0x0000000186975004 0x18692c000 + 299012
6   CoreMotion                    	0x000000018697c018 0x18692c000 + 327704
7   CoreMotion                    	0x0000000186978bb4 0x18692c000 + 314292
8   CoreMotion                    	0x0000000186979ec0 0x18692c000 + 319168
9   CoreMotion                    	0x0000000186979d70 0x18692c000 + 318832
10  CoreMotion                    	0x000000018697f768 0x18692c000 + 341864
11  CoreMotion                    	0x0000000186979d08 0x18692c000 + 318728
12  CoreMotion                    	0x00000001869c1aac 0x18692c000 + 613036
13  CoreMotion                    	0x00000001869c107c 0x18692c000 + 610428
14  CoreMotion                    	0x0000000186979fc8 0x18692c000 + 319432
15  CoreMotion                    	0x0000000186979d70 0x18692c000 + 318832
16  CoreMotion                    	0x000000018697f768 0x18692c000 + 341864
17  CoreMotion                    	0x0000000186979d08 0x18692c000 + 318728
18  CoreMotion                    	0x00000001869c032c 0x18692c000 + 607020
19  CoreMotion                    	0x000000018697f768 0x18692c000 + 341864
20  CoreMotion                    	0x00000001869c0284 0x18692c000 + 606852
21  CoreMotion                    	0x00000001869639a8 0x18692c000 + 227752
22  CoreMotion                    	0x000000018697fa18 0x18692c000 + 342552
23  CoreFoundation                	0x0000000180a90f84 0x1809b0000 + 921476
24  CoreFoundation                	0x0000000180a908bc 0x1809b0000 + 919740
25  CoreFoundation                	0x0000000180a8ed04 0x1809b0000 + 912644
26  CoreFoundation                	0x00000001809b8c50 0x1809b0000 + 35920
27  CoreFoundation                	0x0000000180a063a4 0x1809b0000 + 353188
28  CoreMotion                    	0x000000018697f41c 0x18692c000 + 341020
29  libsystem_pthread.dylib       	0x000000018073fb28 0x18073c000 + 15144
30  libsystem_pthread.dylib       	0x000000018073fa8c 0x18073c000 + 14988
31  libsystem_pthread.dylib       	0x000000018073d028 0x18073c000 + 4136

Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000000127768857
    x4: 0x0000000180136e02   x5: 0x000000016fdb7520   x6: 0x000000000000006e   x7: 0x0000000000000fa0
    x8: 0x0000000008000000   x9: 0x0000000004000000  x10: 0x0000000000000002  x11: 0x0000000000000010
   x12: 0x0000000000000000  x13: 0x0000000000000002  x14: 0x0000000000000000  x15: 0x0000030000000300
   x16: 0x0000000000000148  x17: 0x0000000000000000  x18: 0x0000000000000000  x19: 0x0000000000000006
   x20: 0x000000019ed2b000  x21: 0x000000016fdb7520  x22: 0x0000000127610010  x23: 0x000000019ed37150
   x24: 0x0000000000000001  x25: 0xad0081b2cb3afb14  x26: 0x0000000127611560  x27: 0x0000000000000000
   x28: 0x0000000000000001  fp: 0x000000016fdb7480   lr: 0x0000000180740ef8
    sp: 0x000000016fdb7460   pc: 0x000000018067411c cpsr: 0x00000000

**Comment by shalem93** [comment]
@Raikerian  Have a look when you get the chance^

**Comment by Raikerian** [comment]
@shalem93 We do not have such problems. Our app [Q-municate](https://github.com/QuickBlox/q-municate-ios) is available in App store. The crash you have probably is not Quickblox SDK related. How poor connection can be related to crash in any way? You need to debug it and find where the crash is happening. Setup exception breakpoints for that. You can also add [Crashlytics](https://www.crashlytics.com) to your app, to have a crashlogs available even if app was crashed at Apple. Logs you provided does not saying anything to us.

**Comment by Raikerian** [comment]
@shalem93 do you have any updates on this? Were you able to fix it?

**Comment by shalem93** [comment]
Kind of. 3G and weaker Wi-Fi signals still can't connect to the app. LTE and good Wi-Fi work however. Any idea how we can provide better coverage to our users? @Raikerian

**Comment by Raikerian** [comment]
Hi again,

We have just released our new WebRTC SDK 2.3 which is containing completely new audio module and lots of fixes and improvements.

Here is full changelog of version 2.3: http://quickblox.com/developers/Sample-webrtc-ios#Framework_changelog
You might look into it.

As for issue with connection, were you able to resolve it? You might look into new SDK, if it's still occurs I think we might look into your stat reports again.

Have a nice holidays.
